code/modules/projectiles/gun.dm

```diff
@@ -112,6 +112,7 @@
 		return 0
 	if(!user.IsAdvancedToolUser())
 		return 0
+
 	if(pin && needspin)

```

Comment:
This group of checks should be last in this proc, HULK, CLUMSY, and PACIFIST should be checked first.